RecoBTag/ONNXRuntime/python/SwitchProducerONNX.py

```diff
@@ -0,0 +1,39 @@
+import FWCore.ParameterSet.Config as cms
+
+_tf_enabled_cached = None
+_onnxrt_enabled_cached = None
+
+
+def _switch_native():
+    global _tf_enabled_cached
+    if _tf_enabled_cached is None:
+        import os
+        _tf_enabled_cached = ('CMS_DISABLE_TENSORFLOW' not in os.environ)
+    return (_tf_enabled_cached, 1)
+
+
+def _switch_onnxruntime():
+    global _onnxrt_enabled_cached
+    if _onnxrt_enabled_cached is None:
+        import os
+        _onnxrt_enabled_cached = ('amd64' in os.environ['SCRAM_ARCH'] or 'aarch64' in os.environ['SCRAM_ARCH']) and ('CMS_DISABLE_ONNXRUNTIME' not in os.environ)
+    return (_onnxrt_enabled_cached, 2)
+
+
+class SwitchProducerONNX(cms.SwitchProducer):
+
+    def __init__(self, **kwargs):
+        super(SwitchProducerONNX, self).__init__(
+            dict(native = _switch_native,
+                 onnx = _switch_onnxruntime),
+            **kwargs
+        )
+
+    def cloneAll(self, **params):
```

Comment:
> can't this be called `clone`?

That would override the base class `SwitchProducer.clone()`, which does the clone but requires to explicitly pass the dictionaries for all cases, just like the implementation does. On the other hand see that for the use cases where the case configurations are (nearly) identical, the base `SwitchProducer` approach is a bit cumbersome. I could think of adding a special parameter to the base class `clone()` that the modifications would be applied to all cases, e.g. something along `switchProducer.clone(..., applyToAllCases_ = True)`.

Reply:
> I could think of adding a special parameter to the base class `clone()` that the modifications would be applied to all cases, e.g. something along `switchProducer.clone(..., applyToAllCases_ = True)`.

@makortel I think it would be nice to have this:)